samples/Mvc.RenderViewToString/Program.cs

```diff
@@ -33,25 +42,38 @@ public static void Main()
                 UserData1 = 1,
                 UserData2 = 2
             };
-            var emailContent = helper.RenderViewToString("EmailTemplate", model);
-            Console.WriteLine(emailContent);
-            Console.ReadLine();
+
+            return helper.RenderViewToString("EmailTemplate", model);
         }
 
         private static void ConfigureDefaultServices(IServiceCollection services, string customApplicationBasePath)
         {
             var applicationEnvironment = PlatformServices.Default.Application;
-            services.AddSingleton(applicationEnvironment);
+            string applicationName;
+            IFileProvider fileProvider;
+            if (!string.IsNullOrEmpty(customApplicationBasePath))
+            {
+                applicationName = Path.GetFileName(customApplicationBasePath);
+                fileProvider = new PhysicalFileProvider(customApplicationBasePath);
+            }
+            else
+            {
+                applicationName = applicationEnvironment.ApplicationName;
+                fileProvider = new PhysicalFileProvider(applicationEnvironment.ApplicationBasePath);
+            }
+
             services.AddSingleton<IHostingEnvironment>(new HostingEnvironment
             {
-                WebRootFileProvider = new PhysicalFileProvider(customApplicationBasePath ?? applicationEnvironment.ApplicationBasePath)
+                ApplicationName =  applicationName,
+                WebRootFileProvider = fileProvider,
             });
             services.Configure<RazorViewEngineOptions>(options =>
             {
                 options.FileProviders.Clear();
-                options.FileProviders.Add(new PhysicalFileProvider(customApplicationBasePath ?? applicationEnvironment.ApplicationBasePath));
+                options.FileProviders.Add(fileProvider);
             });
             var diagnosticSource = new DiagnosticListener("Microsoft.AspNetCore");
+            services.AddSingleton<ObjectPoolProvider, DefaultObjectPoolProvider>();
```

Comment:
Since Razor depends on it and we sort of expect it to work in non-hosted scenarios, we should add it as part of `AddRazor` \ `AddMvc`. It's a no-op operation if it already exists via Hosting.